### PR TITLE
Fix broken references in `docs/web_reference.rst`

### PR DESCRIPTION
**Pull request by l1storez**
## What do these changes do?

 This change corrects multiple unrendered intersphinx class reference in the `web_reference.rst` document.

## Are there changes in behavior for the user?

 No

## Related issue number

 #5518 

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by l1storez** [comment]
`aiohttp.web.BaseRequest.read` or `BaseRequest.read`. Should I use prefix `aiohttp.web` or no?

File: docs/web_reference.rst [hunk 1]
```diff
@@ -320,20 +320,20 @@ and :ref:`aiohttp-web-signals` handlers.
 
    .. attribute:: if_match
 
-      Read-only property that returns :class:`ETag` objects specified
+      Read-only property that returns :class:`aiohttp.ETag` objects specified
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      Read-only property that returns :class:`~aiohttp.ETag` objects specified
```

File: docs/web_reference.rst [hunk 2]
```diff
@@ -320,20 +320,20 @@ and :ref:`aiohttp-web-signals` handlers.
...
       in the *If-Match* header.
 
-      Returns :class:`tuple` of :class:`ETag` or ``None`` if
+      Returns :class:`tuple` of :class:`aiohttp.ETag` or ``None`` if
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      Returns :class:`tuple` of :class:`~aiohttp.ETag` or ``None`` if
```

File: docs/web_reference.rst [hunk 3]
```diff
@@ -320,20 +320,20 @@ and :ref:`aiohttp-web-signals` handlers.
...
       *If-Match* header is absent.
 
       .. versionadded:: 3.8
 
    .. attribute:: if_none_match
 
-      Read-only property that returns :class:`ETag` objects specified
+      Read-only property that returns :class:`aiohttp.ETag` objects specified
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      Read-only property that returns :class:`~aiohttp.ETag` objects specified
```

File: docs/web_reference.rst [hunk 4]
```diff
@@ -320,20 +320,20 @@ and :ref:`aiohttp-web-signals` handlers.
...
       *If-None-Match* header.
 
-      Returns :class:`tuple` of :class:`ETag` or ``None`` if
+      Returns :class:`tuple` of :class:`aiohttp.ETag` or ``None`` if
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      Returns :class:`tuple` of :class:`~aiohttp.ETag` or ``None`` if
```

**Comment by webknjaz** [comment]
You should use the full importable-like string in references but you may still opt-in to rendering the short form in an anchor.

File: docs/web_reference.rst [hunk 5]
```diff
@@ -798,7 +798,7 @@ StreamResponse
 
       *ETag* header for outgoing response.
 
-      This property accepts raw :class:`str` values, :class:`ETag`
+      This property accepts raw :class:`str` values, :class:`aiohttp.ETag`
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      This property accepts raw :class:`str` values, :class:`~aiohttp.ETag`
```

**Comment by l1storez** [comment]
Ok, I add "~" to `aiohttp.ETag`

**Comment by l1storez** [comment]
`BaseResource` its `aiohttp.web.Resource` or `aiohttp.web.AbstractResource`?
https://github.com/aio-libs/aiohttp/blame/576862b1f2cb919c347851f40807caf7a9b80807/docs/web_reference.rst#L1814

**Comment by l1storez** [comment]
**Logs/tracebacks after fixes**
> 
> ```
>/aiohttp/docs/web_reference.rst:32: WARNING: py:class reference target not found: aiohttp.protocol.HttpVersion
>/aiohttp/docs/web_reference.rst:421: WARNING: py:class reference target not found: callable
>/aiohttp/docs/web_reference.rst:898: WARNING: py:class reference target not found: aiohttp.payload.StringPayload
>/aiohttp/docs/web_reference.rst:912: WARNING: py:attr reference target not found: StreamResponse.body
>/aiohttp/docs/web_reference.rst:930: WARNING: py:attr reference target not found: body
>/aiohttp/docs/web_reference.rst:1053: WARNING: py:meth reference target not found: start
>/aiohttp/docs/web_reference.rst:1153: WARNING: py:class reference target not found: callable
>/aiohttp/docs/web_reference.rst:1250: WARNING: py:class reference target not found: callable
>/aiohttp/docs/web_reference.rst:1325: WARNING: py:meth reference target not found: Application.copy
>/aiohttp/docs/web_reference.rst:1476: WARNING: py:class reference target not found: MatchedSubAppResource
>/aiohttp/docs/web_reference.rst:1542: WARNING: py:meth reference target not found: asyncio.AbstreactEventLoop.create_server
>/aiohttp/docs/web_reference.rst:1574: WARNING: py:meth reference target not found: router
>/aiohttp/docs/web_reference.rst:1639: WARNING: py:class reference target not found: callable
>/aiohttp/docs/web_reference.rst:1639: WARNING: py:class reference target not found: coroutine
>/aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: PlainRoute
>/aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: DynamicRoute
>/aiohttp/docs/web_reference.rst:1762: WARNING: py:class reference target not found: coroutine
>/aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url
>/aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url_for
>/aiohttp/docs/web_reference.rst:1761: WARNING: py:class reference target not found: StaticRoute
>/aiohttp/docs/web_reference.rst:1813: WARNING: py:class reference target not found: BaseResource
>/aiohttp/docs/web_reference.rst:1952: WARNING: py:class reference target not found: callable
>/aiohttp/docs/web_reference.rst:1952: WARNING: py:class reference target not found: coroutine
>/aiohttp/docs/web_reference.rst:2334: WARNING: py:class reference target not found: collections.Sequence
>/aiohttp/docs/web_reference.rst:2437: WARNING: py:class reference target not found: Route
>/aiohttp/docs/web_reference.rst:2522: WARNING: py:meth reference target not found: socket.getsockname
>/aiohttp/docs/web_reference.rst:2562: WARNING: py:data reference target not found: aiohttp.log.server_logger
>/aiohttp/docs/web_reference.rst:2564: WARNING: py:data reference target not found: aiohttp.log.access_logger
>/aiohttp/docs/web_reference.rst:2566: WARNING: py:data reference target not found: aiohttp.helpers.AccessLogger
>/aiohttp/docs/web_reference.rst:2566: WARNING: py:class reference target not found: aiohttp.abc.AbstractAccessLogger
>/aiohttp/docs/web_reference.rst:2569: WARNING: py:attr reference target not found: helpers.AccessLogger.LOG_FORMAT
>/aiohttp/docs/web_reference.rst:2883: WARNING: py:class reference target not found: socket
>/aiohttp/docs/web_reference.rst:2848: WARNING: py:data reference target not found: aiohttp.helpers.AccessLogger
>/aiohttp/docs/web_reference.rst:2848: WARNING: py:class reference target not found: aiohttp.abc.AbstractAccessLogger
> ```

**Comment by webknjaz** [comment]
> `BaseResource` its `aiohttp.web.Resource` or `aiohttp.web.AbstractResource`?
> [`576862b`/docs/web_reference.rst#L1814 (blame)](https://github.com/aio-libs/aiohttp/blame/576862b1f2cb919c347851f40807caf7a9b80807/docs/web_reference.rst#L1814)

Sounds like the abstract one is the base class so use it (https://github.com/aio-libs/aiohttp/blob/09ac1cbb4c07b4b3e1872374808512c9bc52f2ab/aiohttp/web_urldispatcher.py#L316)

**Comment by l1storez** [comment]
Need help with this:

>aiohttp/docs/web_reference.rst:930: WARNING: py:attr reference target not found: body
>aiohttp/docs/web_reference.rst:1053: WARNING: py:meth reference target not found: start
>aiohttp/docs/web_reference.rst:1542: WARNING: py:meth reference target not found: asyncio.AbstractEventLoop.create_server
>aiohttp/docs/web_reference.rst:1574: WARNING: py:meth reference target not found: router
>aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: PlainRoute
>aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: DynamicRoute
>aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url
>aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url_for
>aiohttp/docs/web_reference.rst:1761: WARNING: py:class reference target not found: StaticRoute
>aiohttp/docs/web_reference.rst:2437: WARNING: py:class reference target not found: Route
>aiohttp/docs/web_reference.rst:2562: WARNING: py:data reference target not found: aiohttp.log.server_logger
>aiohttp/docs/web_reference.rst:2564: WARNING: py:data reference target not found: aiohttp.log.access_logger
>aiohttp/docs/web_reference.rst:2566: WARNING: py:data reference target not found: aiohttp.helpers.AccessLogger
>aiohttp/docs/web_reference.rst:2569: WARNING: py:attr reference target not found: helpers.AccessLogger.LOG_FORMAT
>aiohttp/docs/web_reference.rst:2883: WARNING: py:class reference target not found: socket
>aiohttp/docs/web_reference.rst:2848: WARNING: py:data reference target not found: aiohttp.helpers.AccessLogger

File: docs/conf.py [hunk 6]
```diff
@@ -356,4 +356,7 @@
     ("py:mod", "aiohttp"),  # undocumented, no `.. currentmodule:: aiohttp` in docs
     ("py:class", "aiohttp.SimpleCookie"),  # undocumented
     ("py:class", "aiohttp.web.RequestHandler"),  # undocumented
+    ("py:attr", "StreamResponse.body"),  # undocumented
```
Comment (webknjaz) [inline docs/conf.py]: This should contain the package+module part.

File: docs/conf.py [hunk 7]
```diff
@@ -356,4 +356,7 @@
...
+    ("py:class", "aiohttp.payload.StringPayload"),  # undocumented
+    ("py:meth", "Application.copy"),  # undocumented
```
Comment (webknjaz) [inline docs/conf.py]: This should contain the package+module part too.

File: docs/web_reference.rst [hunk 8]
```diff
@@ -1473,7 +1473,7 @@ duplicated like one using :meth:`Application.copy`.
 
       :param Application subapp: nested application.
 
-      :returns: a :class:`MatchedSubAppResource` instance.
+      :returns: a :class:`PrefixedSubAppResource` instance.
```
Comment (webknjaz) [inline docs/web_reference.rst]: This is wrong, it returns a `MatchedSubAppResource`: https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_app.py#L268

Comment (l1storez) [reply docs/web_reference.rst]: ok, had some qustions about this. `MatchedSubAppResource` goes to ignore

File: docs/web_reference.rst [hunk 9]
```diff
@@ -2519,7 +2519,7 @@ application on specific TCP or Unix socket, e.g.::
 
       A  :class:`list` of served sockets addresses.
 
-      See :meth:`socket.getsockname` for items type.
+      See :meth:`socket.socket.getsockname` for items type.
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      See :meth:`socket.getsockname() <socket.socket.getsockname>` for items type.
```

Comment (l1storez) [reply docs/web_reference.rst]: oh, wow

**Comment by l1storez** [comment]
Remaining warnings, could `staticroute` be `abstractroute`?
>aiohttp/docs/web_reference.rst:930: WARNING: py:attr reference target not found: body
aiohttp/docs/web_reference.rst:1476: WARNING: py:class reference target not found: MatchedSubAppResource
/aiohttp/docs/web_reference.rst:1574: WARNING: py:meth reference target not found: router
/aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: PlainRoute
/aiohttp/docs/web_reference.rst:1638: WARNING: py:class reference target not found: DynamicRoute
/aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url
/aiohttp/docs/web_reference.rst:1754: WARNING: py:meth reference target not found: StaticRoute.url_for
/aiohttp/docs/web_reference.rst:1761: WARNING: py:class reference target not found: StaticRoute
/aiohttp/docs/web_reference.rst:2437: WARNING: py:class reference target not found: Route
/aiohttp/docs/web_reference.rst:2883: WARNING: py:class reference target not found: socket

**Comment by l1storez** [comment]
Forgot to put  `~` before `Application.copy` L1326

File: docs/web_reference.rst [hunk 10]
```diff
@@ -1567,7 +1567,7 @@ Router
 For dispatching URLs to :ref:`handlers<aiohttp-web-handler>`
 :mod:`aiohttp.web` uses *routers*.
 
-Router is any object that implements :class:`AbstractRouter` interface.
+Router is any object that implements :class:`aiohttp.abc.AbstractRouter` interface.
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
Router is any object that implements :class:`~aiohttp.abc.AbstractRouter` interface.
```

File: docs/web_reference.rst [hunk 11]
```diff
@@ -1845,11 +1845,11 @@ unique *name* and at least one :term:`route`.
    finished.
 4. Otherwise router tries next resource from the *routing table*.
 5. If the end of *routing table* is reached and no *resource* /
-   *route* pair found the *router* returns special :class:`AbstractMatchInfo`
-   instance with :attr:`AbstractMatchInfo.http_exception` is not ``None``
+   *route* pair found the *router* returns special :class:`aiohttp.abc.AbstractMatchInfo`
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
   *route* pair found the *router* returns special :class:`~aiohttp.abc.AbstractMatchInfo`
```

File: docs/web_reference.rst [hunk 12]
```diff
@@ -2416,12 +2416,12 @@ Matching result can be accessible from handler as
 :attr:`Request.match_info` attribute.
 
 In general the result may be any object derived from
-:class:`AbstractMatchInfo` (:class:`UrlMappingMatchInfo` for default
+:class:`aiohttp.abc.AbstractMatchInfo` (:class:`UrlMappingMatchInfo` for default
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
:class:`~aiohttp.abc.AbstractMatchInfo` (:class:`UrlMappingMatchInfo` for default
```

File: docs/web_reference.rst [hunk 13]
```diff
@@ -2416,12 +2416,12 @@ Matching result can be accessible from handler as
...
 :class:`UrlDispatcher` router).
 
 .. class:: UrlMappingMatchInfo
 
-   Inherited from :class:`dict` and :class:`AbstractMatchInfo`. Dict
+   Inherited from :class:`dict` and :class:`aiohttp.abc.AbstractMatchInfo`. Dict
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
   Inherited from :class:`dict` and :class:`~aiohttp.abc.AbstractMatchInfo`. Dict
```

File: docs/web_reference.rst [hunk 14]
```diff
@@ -2442,7 +2442,7 @@ View
 
 .. class:: View(request)
 
-   Inherited from :class:`AbstractView`.
+   Inherited from :class:`aiohttp.abc.AbstractView`.
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
   Inherited from :class:`~aiohttp.abc.AbstractView`.
```

**Comment by l1storez** [comment]
It might be easier to write the line numbers without `~`?

File: docs/web_reference.rst [hunk 15]
```diff
@@ -2726,7 +2726,7 @@ application on specific TCP or Unix socket, e.g.::
 
    :param runner: a runner to serve.
 
-   :param sock: :class:`socket.socket` to listen.
+   :param sock: :func:`socket.socket` to listen.
```
Comment (webknjaz) [inline docs/web_reference.rst]: No, this param is not a function, it's a result that this function returns.
There's no label for the class specifically so let's just link "socket objects" section from the docs:
```suggestion
   :param sock: A :ref:`socket instance <socket-objects>` to listen to.
```

File: docs/web_reference.rst [hunk 16]
```diff
@@ -2747,8 +2747,8 @@ Utilities
 
 .. class:: FileField
 
-   A :class:`~collections.namedtuple` instance that is returned as
-   multidict value by :meth:`Request.POST` if field is uploaded file.
+   A :class:`~typing.NamedTuple` instance that is returned as
```
Comment (webknjaz) [inline docs/web_reference.rst]: This is actually a dataclass on master https://github.com/aio-libs/aiohttp/blob/master/aiohttp/web_request.py#L74 and an `attr.s` class on the 3.8 branch.

**Comment by webknjaz** [comment]
> It might be easier to write the line numbers without `~`?

what line numbers?

**Comment by l1storez** [comment]
the warning in L2818?
>web_reference.rst:2883: WARNING: py:class reference target not found: socket

**Comment by l1storez** [comment]
> I need help with `Route`, `PlainRoute`, `DynamicRoute`, coz its "Drop old-style routes"

That doesn't exist anymore, I remove the class role. via 5743

**Comment by webknjaz** [comment]
> > I need help with `Route`, `PlainRoute`, `DynamicRoute`, coz its "Drop old-style routes"
> 
> That doesn't exist anymore, I remove the class role. via 5743

No, that method returns `ResourceRoute` per the source code. Removing the class role will only drop the reference but the text would still point at those names. It's best for the docs to reflect exactly what happens.

**Comment by l1storez** [comment]
>It's best for the docs to reflect exactly what happens.

I agree with that but: 
L1638 :returns: new :class:`PlainRoute` or :class:`DynamicRoute` instance.

**Comment by webknjaz** [comment]
And that information is wrong. It must be corrected.

**Comment by l1storez** [comment]
I understand that. How to make the information correct? Not enough context 😕

**Comment by l1storez** [comment]
Please, check my fixes

File: docs/web_reference.rst (hunk 16 above)
Comment (l1storez) [reply docs/web_reference.rst]: Forget about this, data or attr.s?

Comment (webknjaz) [reply docs/web_reference.rst]: Not data. Dataclass in master and attrs in 3.8.

Comment (webknjaz) [reply docs/web_reference.rst]:
```suggestion
   A :mod:`dataclass <dataclasses>` instance that is returned as
```

File: docs/web_reference.rst [hunk 17]
```diff
@@ -1527,8 +1527,8 @@ duplicated like one using :meth:`Application.copy`.
       based but traversal ones).
 
       For sake of that fact we have very trivial ABC for
-      :class:`AbstractRouter`: it should have only
-      :meth:`AbstractRouter.resolve` coroutine.
+      :class:`aiohttp.abc.AbstractRouter`: it should have only
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      :class:`~aiohttp.abc.AbstractRouter`: it should have only
```

File: docs/web_reference.rst [hunk 18]
```diff
@@ -1754,23 +1754,23 @@ Router is any object that implements :class:`AbstractRouter` interface.
...
 
 
-      :returns: new :class:`StaticRoute` instance.
+      :returns: new :class:`~aiohttp.web.AbstractRoute` instance.
 
    .. comethod:: resolve(request)
 
       A :ref:`coroutine<coroutine>` that returns
-      :class:`AbstractMatchInfo` for *request*.
+      :class:`aiohttp.abc.AbstractMatchInfo` for *request*.
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      :class:`~aiohttp.abc.AbstractMatchInfo` for *request*.
```

File: docs/web_reference.rst [hunk 19]
```diff
@@ -1754,23 +1754,23 @@ Router is any object that implements :class:`AbstractRouter` interface.
...
 
       The method never raises exception, but returns
-      :class:`AbstractMatchInfo` instance with:
+      :class:`aiohttp.abc.AbstractMatchInfo` instance with:
```
Comment (webknjaz) [inline docs/web_reference.rst]:
```suggestion
      :class:`~aiohttp.abc.AbstractMatchInfo` instance with:
```

**Comment by webknjaz** [comment]
Backport #5772

**Pull request merged**